extensions/ql-vscode/src/view/variant-analysis/__tests__/VariantAnalysis.spec.tsx

```diff
@@ -16,6 +18,23 @@ describe(VariantAnalysis.name, () => {
       />,
     );
 
+  const postMessage = async (msg: ToVariantAnalysisMessage) => {
+    await act(async () => {
+      // window.postMessage doesn't set the origin correctly, see
+      // https://github.com/jsdom/jsdom/issues/2745
+      window.dispatchEvent(
+        new MessageEvent("message", {
+          source: window,
+          origin: window.location.origin,
+          data: msg,
+        }),
+      );
+
+      // The event is dispatched asynchronously, so we need to wait for it to be handled.
+      await new Promise((resolve) => setTimeout(resolve, 0));
+    });
+  };
```

Comment:
Now that we're using this in at least two tests, do you think it would make sense to move this to a separate file and export the function?

Reply:
Good idea! I wasn't sure where best to put the shared file. Currently in `src/view/common/post-message.ts` but I'm open to better suggestions 😅